flight/core/Dispatcher.php

```diff
@@ -188,27 +202,38 @@ public function clear(?string $name = null): void
             return;
         }
 
-        $this->events = [];
-        $this->filters = [];
+        $this->reset();
     }
 
     /**
      * Hooks a callback to an event.
      *
      * @param string $name Event name
-     * @param 'before'|'after' $type Filter type
-     * @param Closure(array<int, mixed> &$params, string &$output): (void|false) $callback
+     * @param 'before'|'after' $type Filter type.
+     * @param callable(array<int, mixed> &$params, mixed &$output): (void|false)|callable(mixed &$output): (void|false) $callback
      *
      * @return $this
      */
     public function hook(string $name, string $type, callable $callback): self
     {
-        if (!in_array($type, self::FILTER_TYPES, true)) {
-            $noticeMessage = "Invalid filter type '$type', use " . join('|', self::FILTER_TYPES);
+        static $filterTypes = [self::FILTER_BEFORE, self::FILTER_AFTER];
```

Comment:
It doesn't need to be static does it? Since it's just referring to static types. The other places in the framework that use this are those places trying to avoid using $this->someVar to hold data.

Reply:
Since the only place that used the constant was that function, I moved it to the function, the fact that it is static in the function indicates that it is only initialized once, this is for performance and micro-optimization

https://www.php.net/manual/en/language.variables.scope.php

look for static variables